// ngx-signaturepad/__ivy_ngcc__/index.js
"use strict";
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./signature-pad');
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var signature_pad_1 = require("./signature-pad");
var SignaturePadModule = /** @class */ (function () {
    function SignaturePadModule() {
    }
SignaturePadModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SignaturePadModule });
SignaturePadModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SignaturePadModule_Factory(t) { return new (t || SignaturePadModule)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SignaturePadModule, { declarations: [ɵngcc1.SignaturePad], exports: [ɵngcc1.SignaturePad] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SignaturePadModule, [{
        type: core_1.NgModule,
        args: [{
                imports: [],
                declarations: [signature_pad_1.SignaturePad],
                exports: [signature_pad_1.SignaturePad]
            }]
    }], function () { return []; }, null); })();
    return SignaturePadModule;
}());
exports.SignaturePadModule = SignaturePadModule;
exports.SignaturePad = ɵngcc1.SignaturePad;
