// angular2-signaturepad/__ivy_ngcc__/index.js
"use strict";
var core_1 = require('@angular/core');
var signature_pad_1 = require('./signature-pad');
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./signature-pad');
var SignaturePadModule = (function () {
    function SignaturePadModule() {
    }
    /** @nocollapse */
    SignaturePadModule.ctorParameters = [];
SignaturePadModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SignaturePadModule });
SignaturePadModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SignaturePadModule_Factory(t) { return new (t || SignaturePadModule)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SignaturePadModule, { declarations: [ɵngcc1.SignaturePad], exports: [ɵngcc1.SignaturePad] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SignaturePadModule, [{
        type: core_1.NgModule,
        args: [{
                imports: [],
                declarations: [signature_pad_1.SignaturePad],
                exports: [signature_pad_1.SignaturePad]
            }]
    }], function () { return []; }, null); })();
    return SignaturePadModule;
}());
exports.SignaturePadModule = SignaturePadModule;

exports.SignaturePad = ɵngcc1.SignaturePad;
